src/prefect/settings.py

```diff
@@ -1051,6 +1114,398 @@ class RunnerSettings(PrefectBaseSettings):
+        validation_alias=AliasChoices(
+            AliasPath("port"),
+            "prefect_server_database_port",
+            "prefect_api_database_port",
+        ),
+    )
+
+    user: Optional[str] = Field(
+        default=None,
+        description="The user to use when connecting to the database.",
+        validation_alias=AliasChoices(
+            AliasPath("user"),
+            "prefect_server_database_user",
+            "prefect_api_database_user",
+        ),
+    )
+
+    name: Optional[str] = Field(
+        default=None,
+        description="The name of the Prefect database on the remote server, or the path to the database file for SQLite.",
+        validation_alias=AliasChoices(
+            AliasPath("name"),
+            "prefect_server_database_name",
+            "prefect_api_database_name",
+        ),
+    )
+
+    password: Optional[SecretStr] = Field(
+        default=None,
+        description="The password to use when connecting to the database. Should be kept secret.",
+        validation_alias=AliasChoices(
+            AliasPath("password"),
+            "prefect_server_database_password",
+            "prefect_api_database_password",
+        ),
+    )
+
+    echo: bool = Field(
+        default=False,
+        description="If `True`, SQLAlchemy will log all SQL issued to the database. Defaults to `False`.",
+        validation_alias=AliasChoices(
+            AliasPath("echo"),
+            "prefect_server_database_echo",
+            "prefect_api_database_echo",
+        ),
+    )
+
+    migrate_on_start: bool = Field(
+        default=True,
+        description="If `True`, the database will be migrated on application startup.",
+        validation_alias=AliasChoices(
+            AliasPath("migrate_on_start"),
+            "prefect_server_database_migrate_on_start",
+            "prefect_api_database_migrate_on_start",
+        ),
+    )
+
+    timeout: Optional[float] = Field(
+        default=10.0,
+        description="A statement timeout, in seconds, applied to all database interactions made by the API. Defaults to 10 seconds.",
+        validation_alias=AliasChoices(
+            AliasPath("timeout"),
+            "prefect_server_database_timeout",
+            "prefect_api_database_timeout",
+        ),
+    )
+
+    connection_timeout: Optional[float] = Field(
+        default=5,
+        description="A connection timeout, in seconds, applied to database connections. Defaults to `5`.",
+        validation_alias=AliasChoices(
+            AliasPath("connection_timeout"),
+            "prefect_server_database_connection_timeout",
+            "prefect_api_database_connection_timeout",
+        ),
+    )
+
+    @model_validator(mode="after")
+    def emit_warnings(self) -> Self:
+        """More post-hoc validation of settings, including warnings for misconfigurations."""
+        values = self.model_dump()
+        values = warn_on_database_password_value_without_usage(values)
```

Comment:
I probably did this originally, and not that it should be in this PR, just making a note that we should probably rewrite `warn_on_database_password_value_without_usage` to take `self` and this assignment seems unnecessary